common/flights/src/store_client.rs

```diff
@@ -159,6 +168,43 @@ impl StoreClient {
         anyhow::bail!("invalid response")
     }
 
+    pub async fn scan_partition(
+        &mut self,
+        db_name: String,
+        tbl_name: String,
+        scan_plan: &ScanPlan,
+    ) -> anyhow::Result<ScanPartitionResult> {
+        let mut plan = scan_plan.clone();
+        plan.schema_name = format!("{}/{}", db_name, tbl_name);
+        let action = StoreDoAction::ScanPartition(ScanPartitionAction { scan_plan: plan });
+        let rst = self.do_action(&action).await?;
+
+        if let StoreDoActionResult::ScanPartition(rst) = rst {
+            return Ok(rst);
+        }
+        anyhow::bail!("invalid response")
+    }
+
+    /// Get partition.
+    pub async fn read_partition(
+        &mut self,
+        schema: SchemaRef,
+        read_action: &ReadAction,
```

Comment:
A little bit of inconsistency: `scan_partition` use a `ScanPlan` as input arg while `read_partition` accepts an `Action` as input arg.

Reply:
Yeah. Currently, there is no PlanNode passed into ITable::read yet, maybe we can improve this when https://github.com/datafuselabs/datafuse/issues/640 is resolved

Reply:
Ok leave it as is